packages/typescriptlang-org/src/templates/pages/dt/search.tsx

```diff
@@ -0,0 +1,254 @@
+          autoFocus
+          className="searchInput"
+          onChange={event => setSearch(event.target.value)}
+          placeholder="Type Search"
+          type="search"
+          value={search}
+        />
+        {search === results?.query && (
+          <span aria-live="polite" className="resultsCount">
+            {pluralize(results.hits.length, 50, "match", "es")}
+          </span>
+        )}
+        {!search ? (
+          <img
+            alt=""
+            className="magnifier"
+            src={withPrefix("images/dt/magnifier.svg")}
+          />
+        ) : (
+          <input
+            alt="Clear search"
+            className="closer"
+            onClick={() => setSearch("")}
+            src={withPrefix("images/dt/close.svg")}
+            type="image"
+          />
+        )}
+      </div>
+      {!search && (
+        <p className="description">
+          Find npm packages that have type declarations, either bundled or on
+          DefinitelyTyped.
+        </p>
+      )}
+    </div>
+  )
+}
+
+const pluralize = (count: number, max: number, text: string, suffix: string) =>
+  count > max
+    ? `>${max} ${text}${suffix}`
+    : `${count} ${text}${count === 1 ? "" : suffix}`
+
+type SearchResultsProps = {
+  results?: SearchResults
+  search: string
+}
+
+const SearchResultsDisplay: React.FC<SearchResultsProps> = ({
+  results,
+  search,
+}) => {
+  if (!results) {
+    return <div>default search goes here ;)</div>
+  }
+
+  if (!results.hits.length) {
+    return <div>sad, no results for {search} :(</div>
+  }
+
+  console.log(results.hits.map(hit => hit.name))
+
+  return (
+    <table>
+      <tbody>
+        {results.hits.map(hit => (
+          <HitResult
+            definitelyTyped
+            key={hit.name}
+            hit={hit}
+            installer={["npm i", "--save-dev"]}
+          />
+        ))}
+      </tbody>
+    </table>
+  )
+}
+
+type HitResultProps = {
+  definitelyTyped?: boolean
+  hit: SearchHit
+  installer: [string, string]
+}
+
+const HitResult: React.FC<HitResultProps> = ({
+  definitelyTyped,
+  hit,
+  installer,
+}) => {
+  return (
+    <tr className="hitResult">
+      <td
+        className={cx(
+          "hitDownloads",
+          hit.downloadsLast30Days > 1_000_000
+            ? "hitDownloadsMillion"
+            : hit.downloadsLast30Days > 100_000
+            ? "hitDownloadsHundredThousand"
+            : "hitDownloadsMeh"
+        )}
+      >
+        {hit.humanDownloadsLast30Days}
+      </td>
+      <td>??</td>
+      <td>
+        <strong>{hit.name}</strong> <br /> {hit.description}
+      </td>
+      <td>{hit.modified}</td>
+      <td>
+        <pre>
+          <code>
+            &gt; {installer[0]} {hit.name}
+            {definitelyTyped && (
+              <>
+                {"\n"}
+                &gt; {installer[0]} @types/${hit.name} {installer[1]}
+              </>
+            )}
+          </code>
+        </pre>
+      </td>
+    </tr>
+  )
+}
```

Comment:
This page is pretty huge and only getting bigger. Do you think it's time for moving some of these into a `components/` directory?

Reply:
Update: I couldn't resist...